### ISSUE TITLE
Package.json takes half of package size once bundled

### ISSUE THREAD

**Issue by MrCrayon** [OpenVidu/openvidu#488]
**Describe the bug**
openvidu-browser imports/require package.json to get version number, that means that when all modules are bundled together package.json is included almost doubling the size required from openvidu-browser (see screenshot).
https://github.com/OpenVidu/openvidu/blob/1983fc87fc50c8a59c7f317c78eb7b47094bafc6/openvidu-browser/src/OpenVidu/OpenVidu.ts#L53
There are other files that require package.json.

**Expected behavior**
openvidu-browser should not import package.json and maybe set version number in another file, I'm afraid I'm not sure what's the best practice

**Wrong current behavior**
openvidu-browser imported in another project is bringing a lot of unneeded json data into production bundle.

**OpenVidu tutorial where to replicate the error**
I know it's important to give an example with tutorials but I think they all use openvidu-browser already bundled so I'm not sure how to modify those project to replicate the problem, I apologize for that.

I think the problem should already be clear and you are probably more skilled than me using npm packages but this is the minimum setup I came up with to replicate:
(I use Vue because that's what I'm using but it's probably the same with other setups)

1. `npm install -g vue-cli`
2. `vue create openvidu-test`
3. select default
4. `cd openvidu-test`
5. `npm run build` at the end it will show the size of chunk-vendors 89.18 KiB
6. `npm install openvidu-browser`
7. Edit src/app.js and add
```javascript
import { OpenVidu } from 'openvidu-browser'
Vue.prototype.$OV = OpenVidu
```
8. run again `npm run build` now chunk-vendors is 377.63 KiB half of the additional size is from package.json.

**Optional to analyze js bundle**
1. `npm install --save-dev webpack-bundle-analyzer`
2. add a vue.config.js in root folder with:
```javascript
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer')
    .BundleAnalyzerPlugin;

module.exports = {
  configureWebpack: {
    plugins: [new BundleAnalyzerPlugin()]
  }
};
```
3. run again `npm run build --report` 

**OpenVidu deployment info**
not relevant

**Client device info (if applicable)**
not relevant

**Screenshots**
![Schermata da 2020-06-02 09-52-29](https://user-images.githubusercontent.com/6875243/83500701-68ab7f80-a4bf-11ea-9b79-5fe58e7b4df6.png)

**Additional context**
Please let me know if something is not clear, I'm no expert in bundling npm packages, using webpack etc. so I might have missed some important information.

**Comment by pabloFuente** [comment]
Hi!

Thanks for your feedback. We will take a look to this and come back with a beta release of the client library to see if the problem is mitigated. I've a couple of theories about what may be the cause of this issue. Maybe it has nothing to do with accesing the package.json in the code, but with a configuration property of the library that is affecting the compilation procedure. Any NPM dependency has its own package.json, as you can check if you take a look to the node_modules folder. But it is true that a common package.json ususally takes 2 to 3KB, not 150KB. Nonetheless, if it is in fact the access to the package.json what is causing the issue, we have already a proposal to have any kind of environment variable (such as the version number) in a separate file.

Regards.

**Comment by MrCrayon** [comment]
Thanks for looking it up.
In bundled `openvidu-browser-2.14.0.min.js` `package.json` is also included but it is still in it's _original_ state so not much of a problem.

Instead when using openvidu-browser with npm at install package.json is changed to reflect all packages resolved with actual version and integrity checks transforming a 1KB file into a 200KB fat one.

As a workaround I noticed that if I override `package.json` after install with the original one build is still successful, probably because `npm-shrinkwrap.json` is also generated and has priority.

**Comment by wardhanster** [comment]
+1 just came here to say I also noticed that.
I hope this is addressed soon enough

**Comment by MrCrayon** [comment]
Just to keep issue info up to date I wanted to confirm that 2.15.0 is still affected.

**Comment by pabloFuente** [comment]
Yes, this is the first thing we will work on after finally publishing the release. And we will publish a compatible beta when fixed.

Regards.

**Comment by pabloFuente** [comment]
We've released **openvidu-browser@2.15.1-alpha1**. By following your instructions using a brand new Vue application, I get the following results:

```
File                                 Size               Gzipped

// No openvidu-browser
dist/js/chunk-vendors.b0f460c7.js    89.18 KiB          31.93 KiB

// openvidu-browser@2.15.0
dist/js/chunk-vendors.b96397a1.js    379.39 KiB         114.89 KiB

// openvidu-browser@2.15.1-alpha1
dist/js/chunk-vendors.9340c684.js    254.81 KiB         75.95 KiB
```

Seems like the problem was related to the fact that the library was forcing fixed versions of package.json dependencies with `npm shrinkwrap`. Changing to common package-lock.json strategy reduces the size of the final compilation to a more reasonable 254 KB.

This is the analysis of `webpack-bundle-analyzer` for 2.15.0 and 2.15.1-alpha1:

---
Before fix (2.15.0)
![image (6)](https://user-images.githubusercontent.com/15049534/86901803-64513280-c10d-11ea-82f1-3b4af3bb6d8e.png)

After fix (2.15.1-alpha1)
![image (7)](https://user-images.githubusercontent.com/15049534/86901850-7337e500-c10d-11ea-9afc-93ac962f7f03.png)

---

2.15.0

![image (9)](https://user-images.githubusercontent.com/15049534/86902640-87301680-c10e-11ea-88db-37935b0815b9.png)

2.15.1-alpha1

![image (10)](https://user-images.githubusercontent.com/15049534/86902800-bcd4ff80-c10e-11ea-82d5-c9da1299fae1.png)

Please, give 2.15.1-alpha1 a try and let me know the results.

Cheers!

**Comment by MrCrayon** [comment]
I can confirm it solved the problem :champagne: 

Thank you very much!

I'm closing this issue since fix is already merged, hope that's OK with your workflow, I'm more familiar with: PR reference issue and when it's merged autocloses it.
